docs/DEVGUIDE.md

```diff
@@ -18,6 +28,26 @@ layout:
     ├── pkg            # Contains all non-"main" Go packages
     └── vendor         # Glide-managed dependencies (untracked)
 
+## Working on Issues
+
+Github does not allow non-maintainers to assign, or be assigned to, issues.
+As such non-maintainers can indicate their desire to work on (own) a particular
+issue by adding a comment to it of the form:
+
+	#dibs
+
+However, it is a good idea to discuss the issue, and your intent to work on it,
+with the other members via the slack channel to make sure there isn't some
+other work alread going on with respect to that issue.
```

Comment:
> As a newbie telling me to ping someone (how I know who is a mystery) and ask them to take time from their busy day to ride side-saddle with me on what could be some trivial issue/PR sounds like a roadblock to encouraging new people to join our group.

I think that makes my proposal sound _way_ more onerous than it actually is.

Put yourself, for the moment, in the shoes of someone contributing to _any_ given project on github for the first time. If you see an issue that needs working on, chances are very, very good that through a comment, you are going to express your willingness to work on the issue. I know I do, for the simple fact that I don't want my efforts to overlap with someone else's. I expect people wishing to contribute to our project (who do not have the luxury / ability to merely assign to themselves) will similarly express their willingness or intent to claim an issue.

So, can we assume that all of the above just happens naturally and that it's equivalent to calling "dibs" without specifically requiring the use of that word?

The _only_ bit of process that is added by what I have proposed is that when this happens, we (the maintainers) designate one of ourselves to assist the newcomer. This doesn't mean pairing with them. It means taking an interest in the PR, fielding a few comments, and being among its reviewers. This doesn't sound like too much process to me, but it does ensure that when a newcomer has claimed some work that a maintainer is still accountable for it. This is important in cases where someone calls dibs and then drops off the face of the earth and work needs to be re-assigned, or in cases where someone calls dibs and then encounters technical hurdles that necessitate either assistance or re-assignment.